client/src/pages/CreatePokemon.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useRef } from 'react';
+import CanvasDraw from 'react-canvas-draw';
+import './style.css';
+import API from '../API';
+import Navbar from '../components/Navbar';
+import Type from '../components/Type';
+
+const CreatePokemon = () => {
+    const [name, setName] = useState('');
+    const [color, setColor] = useState('#000');
+    const canvasInput = useRef();
+
+    /**
+     * Method to handle the creation of a Pokemon on form submit.
+     */
+    const handleCreatePokemon = (e) => {
+        e.preventDefault();
+
+        const payload = {
+            name: e.target.name.value,
+            desc: e.target.desc.value,
+            image: canvasInput.current.getSaveData(),
+            type1: e.target.type1.value,
+            type2: e.target.type2.value,
+            moves: [
+                {
+                    name: e.target.move1.value,
+                    type: e.target.move1type.value,
+                    power: e.target.move1power.value
+                },
+                {
+                    name: e.target.move2.value,
+                    type: e.target.move2type.value,
+                    power: e.target.move2power.value
+                },
+                {
+                    name: e.target.move3.value,
+                    type: e.target.move3type.value,
+                    power: e.target.move3power.value
+                },
+                {
+                    name: e.target.move4.value,
+                    type: e.target.move4type.value,
+                    power: e.target.move4power.value
+                }
+            ]
+        };
+        console.log(payload);
+        // API.createPokemon(payload);
+    };
```

Comment:
calling the API should work now so you can go ahead and uncomment and get that going

Reply:
API call is going through, but it gives me a bad request when type2 isn't filled in. This shouldn't be the case, since I'm passing in an empty string, which should be okay? It seems to be fine if I leave the move types empty, so I'm assuming it has to do with maybe type2 being required in the backend?

Reply:
Just fixed it and it works now